File: school/summer2013/example_openmp/c/02_data_scope.c
#include <stdio.h>
#include <omp.h>

int main()
{
	int i=10, tid;

#pragma omp parallel private(tid) firstprivate(i)
{
	tid = omp_get_thread_num();
	printf("tid = %d i=%d\n", tid, i);
	i = 20;
}
	printf("tid = %d i=%d\n", tid, i);
}

File: summer2013/openmp/fortran/02_data_scope.f90
program data_scope_firstprivate
    integer i, tid, omp_get_thread_num

    i = 10
    call omp_set_num_threads(4)
!$omp parallel private(tid) firstprivate(i)
    tid = omp_get_thread_num()
    print *, ' tid = ', tid, ' i = ', i
    i = 20
!$omp end parallel
    print *, ' tid = ', tid, ' i = ', i
end
